### ISSUE TITLE
Coords to txt file

### ISSUE THREAD

**Issue by BartoszJedynak** [kolton/d2bot-with-kolbot#1252]
Hi,
is possible to add hotkey who will be write current coords? it Will be useful when changing attacks in diablo/baal script.
Best regards
Tom

**Comment by drogga** [comment]
me.x or me.y in Console will show you the current coordinates I think

**Comment by DarkHorseDre** [comment]
I think a hotkey would be a d2manager feature?

for coords you can do something that @gzer told me: `while(true){say(me.x+","+me.y);delay(2000);}`

change `say` for something like `D2Bot.printToConsole("",n)` if you want it in the manager console

**Comment by 5noop** [comment]
It's more of a dev feature so I'm not sure if it should be added, but it's possible to edit or add hotkeys in ToolsThead.js
Alternatively you can use mine https://pastebin.com/YDGA3wae but at your own risk, hotkeys may be different and I'm not sure if I did modify other stuff. Anyway with this you can press f8 and it will display coordinates on the bottom left.

**Comment by BartoszJedynak** [comment]
thx for answer but i need the coords to write to txt file

**Comment by DarkHorseDre** [comment]
You can adapt what @5noop wrote to write to a file instead of display.  

I dont know the exact code but looking at misc.js it could be something like:
`this.fileAction("logs/Coords.txt", 2, "x:" + me.x.toString() + " y:" + me.y.toString() + "\n");`

**Comment by BartoszJedynak** [comment]
i'll write something like this:
FileTools.writeText("Logs/Coords/" +me.charname + "-coords.txt" ,  "New Coord:(" + Pather.getAreaName(me.area) + ", " + me.x + ", " + me.y + ")" );

it's write coords to char ale file eq mycharename-coords.txt

**Comment by 5noop** [comment]
> Now i need add date before area name

`New Date().toString()` will give you the current date

**Comment by BartoszJedynak** [comment]
> > Now i need add date before area name
> 
> `New Date().toString()` will give you the current date

not `" + d.getDate + "`?

**Comment by DarkHorseDre** [comment]
getdate is day of month, and get full year yyyy.
I did this for my itemlog.txt to get internatikonal date format (month starts at 0 so...):
dateString = "[" **+ yyyy + (mm < 10 ? "0" + mm : mm) + (dd < 10 ? "0" + dd : dd) + " - "** + (h < 10 ? "0" + h : h) + ":" + (m < 10 ? "0" + m : m) + ":" + (s < 10 ? "0" + s : s) + "]";

**Comment by BartoszJedynak** [comment]
that will be correct?

`FileTools.writeText("Logs/Coords/" +me.charname + "-coords.txt" , "New Coord:([" + date.getHours() +":" + date.getMinutes() + ":" + date.getSeconds() +"][" + d.getDate ()+ "-" + d.getMonth()+ "-" + d.getFullYear() + "][" + Pather.getAreaName(me.area) + "][" + me.x + ", " + me.y + "])" );`

**Comment by DarkHorseDre** [comment]
yeah sorry, i declared mine in the format of the existing function (build a string then use) similar to you, as:

date = new Date(),
h = date.getHours(),
m = date.getMinutes(),
s = date.getSeconds(),
yyyy = date.getFullYear(), mm = (date.getMonth() + 1), dd = date.getDate(),
dateString = "[" **+ yyyy + (mm < 10 ? "0" + mm : mm) + (dd < 10 ? "0" + dd : dd) + " - "** + (h < 10 ? "0" + h : h) + ":" + (m < 10 ? "0" + m : m) + ":" + (s < 10 ? "0" + s : s) + "]";

**Comment by 5noop** [comment]
> not `" + d.getDate + "`? or ` " + d.getFullYear + "`?

No need except if you don't like the default formatting, then you can do something like @DarkHorseDre did.

**Comment by DarkHorseDre** [comment]
Yeah, my needs were to have the full date and time at start of line so I could scan every line for specific dates easily.
I set the yyyymmddhhmmss so that the dates were ordered by date first (yours looks like:
`"New Coord: [hh:mm:ss][dd-mm-yyyy][areaname][xxxxxxx,yyyyyy]"`

you may want to consider starting with yyyymmdd  so the ordering is by full date and not starting by day and time, e.g.

`"New coord: [2019-02-21][12:45:11][Blood Moor][7025,1048]"`
`"New coord: [2019-02-21][12:45:11][Blood Moor][7025,1048]"`
`"New coord: [2019-02-21][12:46:13][Blood Moor][7045,1011]"`
`"New coord: [2019-02-22][18:45:11][Blood Moor][7321,1076]"`
`"New coord: [2019-02-23][19:21:11][Blood Moor][7987,1148]"`

it makes scanning by eye easier

**Comment by BartoszJedynak** [comment]
> Yeah, my needs were to have the full date and time at start of line so I could scan every line for specific dates easily.
> I set the yyyymmddhhmmss so that the dates were ordered by date first (yours looks like:
> `"New Coord: [hh:mm:ss][dd-mm-yyyy][areaname][xxxxxxx,yyyyyy]"`
> 
> you may want to consider starting with yyyymmdd so the ordering is by full date and not starting by day and time, e.g.
> 
> `"New coord: [2019-02-21][12:45:11][Blood Moor][7025,1048]"`
> `"New coord: [2019-02-21][12:45:11][Blood Moor][7025,1048]"`
> `"New coord: [2019-02-21][12:46:13][Blood Moor][7045,1011]"`
> `"New coord: [2019-02-22][18:45:11][Blood Moor][7321,1076]"`
> `"New coord: [2019-02-23][19:21:11][Blood Moor][7987,1148]"`
> 
> it makes scanning by eye easier
can u give me all your code for coords?

**Comment by DarkHorseDre** [comment]
yeah I think so - but dont change the way youre writing yours to mimick what I did (from misc.js exisitng code) - if what you are writing suits your needs go for it. you have a single line log line versus misc.js function to build the log line then print it.. horses for courses brah
